### PR TITLE
Added committed offsets in KafkaTopicPartitionTracker and diag endpoint response

### PR DESCRIPTION
**Pull request by jzakaryan**
Important: DO NOT REPORT SECURITY ISSUES DIRECTLY ON GITHUB.  
For reporting security issues and contributing security fixes,  
please, email security@linkedin.com instead, as described in  
the contribution guidelines.

Please, take a minute to review the contribution guidelines at:  
https://github.com/linkedin/Brooklin/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionTracker.java [hunk 1]
```diff
@@ -32,8 +33,8 @@
   private final String _consumerGroupId;
 
   private final Map<String, Set<Integer>> _topicPartitions = new ConcurrentHashMap<>();
-
-  private final Map<String, Map<Integer, Long>> _consumerOffsets = new ConcurrentHashMap<>();
+  private final Map<String, Map<Integer, Long>> _consumedOffsets = new ConcurrentHashMap<>();
```
Comment (jzakaryan) [inline datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionTracker.java:36]: I was considering adding a separate `ConsumerOffsets` class to hold both consumed offset and committed offset and use that in the nested dictionary. But synchronizing the writes to the nested dictionaries was a not easy. `onPartitionsPolled` and `onOffsetsCommitted` calls would hold locks on the dictionaries and it would make this code bug prone and non-efficient. So I thought it would be better for the client-side to merge these dictionaries (if necessary for presentation purposes).

File: datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionTracker.java [hunk 2]
```diff
@@ -75,22 +76,30 @@ public void onPartitionsRevoked(@NotNull Collection<TopicPartition> topicPartiti
...
     // here is that another host may handle these partitions due to rebalance, and we don't want to have duplicate
     // consumer offsets for affected partitions (even though the ones with larger offsets wins).
+    removeOffsetsForTopicPartition(topicPartitions, _consumedOffsets);
+
+    // Remove committed offsets for partitions that have been revoked.
+    removeOffsetsForTopicPartition(topicPartitions, _committedOffsets);
+  }
+
+  private void removeOffsetsForTopicPartition(@NotNull Collection<TopicPartition> topicPartitions,
+      Map<String, Map<Integer, Long>> committedOffsets) {
```
Comment (vishwajith-s) [inline datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionTracker.java:89]: Nitpick. May be you need to call this just offsets since its being used by both _consumedOffsets and _committedOffsets

**Review by vishwajith-s — APPROVED** [review]

File: datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionTracker.java (hunk 2 above)
Comment (somandal) [reply datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionTracker.java:89]: +1 make this generic. We also intend to add end offsets to this list.

**Review by somandal — APPROVED** [review]
Was just wondering if it makes sense to add a test where flushless is not enabled, so that we can validate that "committedOffsets" is empty but "consumedOffsets" isn't, and it doesn't break anything?
